### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by crypto-facs**
## Description

This PR aims to migrate most of the RPC tests into the integration tests, not add new tests in order to improve our current test coverage. 
We are currently using geth Client interface to communicate with the RPC endpoints, because of this not all RPC tests were possible to migrate. In order to support tests for all of our endpoints we might have to consider creating a similar client interface for Ethermint's RPC endpoints.
This PR succesfully migrates and adds a couple improvements to the following endpoint's test:
- eth_getTransactionCount
- eth_getBlockTransactionCountByHash
- eth_getTransactionByBlockHashAndIndex
- eth_blockNumber
- eth_getBalance
- eth_getStorageAt
- eth_getCode
- eth_sendTransaction
- eth_estimateGas
- eth_getTransactionReceipt
- eth_getBlockBynumber
- eth_getLogs
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: tests/e2e/integration_test.go [hunk 1]
```diff
@@ -72,6 +75,7 @@ func (s *IntegrationTestSuite) SetupSuite() {
 	s.Require().NoError(err)
 	s.gethClient = gethclient.New(rpcClient)
 	s.Require().NotNil(s.gethClient)
+	s.ethSigner = ethtypes.LatestSignerForChainID(nil)
```
Comment (fedekunze) [inline tests/e2e/integration_test.go:78]: you can pass the chain ID here

**Comment by prajjwol** [comment]
this is awesome!

**Comment by crypto-facs** [comment]
Old tests are removed now, just left the ones that have not been migrated yet for reference.

**Review by fedekunze — APPROVED** [review]
🎉💯🎊

**Pull request merged**